_posts/2018-10-19-Update-on-crates.io-incident.md

```diff
@@ -0,0 +1,133 @@
+
+## Action Taken
+
+The user's IP address was banned immediately. We then backdated the users' packages to remove
+their packages from the homepage. We also redirected the `cratesio` user's page to a 404.
+
+Finally, the `cratesio` user, and all crates they uploaded were deleted.
+The user was reported to GitHub, and has since been banned by them.
+
+## Timeline of events
+
+- 20:09 UTC: The GitHub user `cratesio` registers an account
+- 20:13 UTC: This user begins uploading packages at a rate of roughly one package
+  every 2 seconds
+- 20:17 UTC: All requests updating the index begin to take 10+ seconds
+- 20:41 UTC: An email is sent to the Rust moderation team reporting this user
+- 20:46 UTC: The report is forwarded to the crates.io team
+- 20:50 UTC: The user is reported in the crates.io team Discord.
+- 21:00 UTC: The user's IP address is blocked from accessing the site
+- 21:20 UTC: The user's packages were removed from the crates.io homepage
+- 21:20 UTC: The incident is announced on status.crates.io
+- 22:49 UTC: The user's account page on crates.io is removed.
+- 23:58 UTC: The packages, all associated data, and the user's account are deleted
+  from crates.io
+- 00:40 UTC: The packages are removed from the index.
+
+## Future measures
+
+It should not have been possible for a single user or IP address to upload that
+many packages in a short period of time. We will be introducing rate limiting on
+this endpoint to limit the number of packages a script is able to upload in the
+future.
+
+We are also looking into disallowing usernames that could be impersonating
+official Rust teams. We will be updating our policies to clearly state that this
+form of impersonation is not allowed. We will be deciding the exact wording of
+this policy in the coming weeks.
+
+While it is impossible to tell a user's intent, many, including the team, have
+speculated that this action was either associated with or directly related to the
+recent esclation in community frustration around crates.io policies, in particular,
+the squatting policy.
+
+Regardless of whether this incident had this intent, the cratesio team would like
+to reiterate that taking actions such as the one we experienced on Tuesday is not
+an appropriate way nor effective way to contribute to dialogue about crates.io policy.
+We will be adding a policy making it clear that attempting to disrupt crates.io in order
+to make or further a point is not approrpriate and will be considered a malicous attack.
+We will be deciding on the exact wording of this policy in the coming weeks.
+
+If you feel that a policy is problematic, the correct place to propose a change is by
+creating an RFC or messaging the team at help@crates.io. 
+
+We also have seen a lot of frustration that we are not listening to the concerns
+that are being raised on both official and unofficial Rust forums. We agree that we should
+improve our communication with the community and intend to develop more processes
```

Comment:
Probably good to clarify who "we"/"our" is here, e.g. by mentioning "the crates.io team" somewhere